hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+        assertRowsWithType("SELECT JSON_VALUE(this, '$.stringField' RETURNING VARCHAR) FROM test" ,
+                singletonList(SqlColumnType.VARCHAR), rows(1, "6"));
+        assertRowsWithType("SELECT JSON_VALUE(this, '$.charField' RETURNING VARCHAR) FROM test" ,
+                singletonList(SqlColumnType.VARCHAR), rows(1, "7"));
+        assertRowsWithType("SELECT JSON_VALUE(this, '$.floatField' RETURNING REAL) FROM test" ,
+                singletonList(SqlColumnType.REAL), rows(1, 8.0f));
+        assertRowsWithType("SELECT JSON_VALUE(this, '$.doubleField' RETURNING DOUBLE) FROM test" ,
+                singletonList(SqlColumnType.DOUBLE), rows(1, 9.0));
+    }
+
+    @Test
+    public void when_extendedSyntaxIsSpecified_queryWorksCorrectly() {
+        final IMap<Long, HazelcastJsonValue> test = instance().getMap("test");
+        test.put(1L, new HazelcastJsonValue(""));
+        test.put(2L, new HazelcastJsonValue("[1,2,"));
+        execute("CREATE MAPPING test TYPE IMap OPTIONS ('keyFormat'='bigint', 'valueFormat'='json')");
+
+        assertThrows(HazelcastSqlException.class, () -> query("SELECT JSON_VALUE(this, '$' ERROR ON EMPTY) AS c1 FROM test WHERE __key = 1"));
+        assertNull(querySingleValue("SELECT JSON_VALUE(this, '$' NULL ON EMPTY) AS c1 FROM test WHERE __key = 1"));
+        assertEquals((byte) 1, querySingleValue("SELECT JSON_VALUE(this, '$' DEFAULT 1 ON EMPTY) AS c1 FROM test WHERE __key = 1"));
+
+        assertThrows(HazelcastSqlException.class, () -> query("SELECT JSON_VALUE(this, '$' ERROR ON ERROR) FROM test WHERE __key = 2"));
+        assertNull(querySingleValue("SELECT JSON_VALUE(this, '$' NULL ON ERROR) FROM test WHERE __key = 2"));
+        assertEquals((byte) 1, querySingleValue("SELECT JSON_VALUE(this, '$' DEFAULT 1 ON ERROR) AS c1 FROM test WHERE __key = 2"));
+    }
+
+    @Test
+    public void when_arrayIsReturned_errorIsThrown() {
+        final IMap<Long, HazelcastJsonValue> test = instance().getMap("test");
+        test.put(1L, new HazelcastJsonValue("[1,2,3]"));
+        execute("CREATE MAPPING test TYPE IMap OPTIONS ('keyFormat'='bigint', 'valueFormat'='json')");
+
+        assertThrows(HazelcastSqlException.class, () -> query("SELECT JSON_VALUE(this, '$' ERROR ON ERROR) FROM test"));
+        assertNull(querySingleValue("SELECT JSON_VALUE(this, '$' NULL ON ERROR) FROM test"));
+        assertEquals((byte) 1, querySingleValue("SELECT JSON_VALUE(this, '$' DEFAULT 1 ON ERROR) FROM test"));
+    }
+
+    @Test
+    public void when_objectIsReturned_errorIsThrown() {
+        final IMap<Long, HazelcastJsonValue> test = instance().getMap("test");
+        test.put(1L, new HazelcastJsonValue("{\"test\":1}"));
+        execute("CREATE MAPPING test TYPE IMap OPTIONS ('keyFormat'='bigint', 'valueFormat'='json')");
+
+        assertThrows(HazelcastSqlException.class, () -> query("SELECT JSON_VALUE(this, '$' ERROR ON ERROR) FROM test"));
```

Comment:
This is too wide-catching. It can be any kind of `HazelcastSqlException` and the test will pass, including, for example, SQL parsing eror. We need to assert the message too, or at least a substantial part of it. This way we won't ensure just that what should fail fails, but also that the error message is sensible; and in this specific case the error msg is 
```
Execution on a member failed: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{06d1-66ab-fec0-0001/Project#0}: com.hazelcast.sql.impl.QueryException: JSON_VALUE failed: 
```
```suggestion
        assertThatThrownBy(() -> query("SELECT JSON_VALUE(this, '$' ERROR ON ERROR) FROM test"))
                .isInstanceOf(HazelcastSqlException.class)
                .hasMessageContaining("Result of JSON_VALUE can not be an array or object");
```